File: typescript/src/stats.ts
export class PingStats {
  private nbSuccesses: number = 0;
  private nbFails: number = 0;

  addSuccess() {
    this.nbSuccesses += 1;
  }

  addFail() {
    this.nbFails += 1;
  }

  successRate(): number {
    if (this.total() === 0) {
      return 0;
    }

    return this.nbSuccesses / this.total();
  }

  total(): number {
    return this.nbSuccesses + this.nbFails;
  }
}

File: python/ping/stats.py
class PingStats:
    def __init__(self) -> None:
        self._nb_successes = 0
        self._nb_fails = 0

    def add_success(self):
        self._nb_successes += 1

    def add_fail(self):
        self._nb_fails += 1

    @property
    def total(self) -> int:
        return self._nb_successes + self._nb_fails

    @property
    def success_rate(self) -> float:
        if self.total == 0:
            return 0.0

        return self._nb_successes / self.total
